handler/donation.py

```diff
@@ -0,0 +1,121 @@
+from flask import jsonify
+
+from dao.donation import Donation
+from handler.user import BaseHandler
+
+
+class DonationHandler(BaseHandler):
+    @staticmethod
+    def get_all_donations(available=None):
+        try:
+            donations = Donation.get_all_donations() if not available else Donation.get_available_donations()
+            donation_list = [donation.to_dict() for donation in donations]
+            result = {
+                "message": "Success!",
+                "donations": donation_list,
+            }
+            return jsonify(result), 200
+        except:
+            return jsonify(message="Server error!"), 500
+
+    @staticmethod
+    def get_all_donations_without_request():
+        try:
+            donations = Donation.get_all_donations_without_request()
+            donation_list = [donation.to_dict() for donation in donations]
+            result = {
+                "message": "Success!",
+                "donations": donation_list,
+            }
+            return jsonify(result), 200
+        except:
+            return jsonify(message="Server error!"), 500
+
+    @staticmethod
+    def get_donation_by_id(did, relationship=None):
+        if did:
+            try:
+                donation = Donation.get_donation_by_id(donation_id=did)
+                if not donation:
+                    return jsonify(message="Not Found!"), 404
+                else:
+                    result = {"message": "Success!"}
+                    if relationship:
+                        result['requsts'] = donation.get_all_donation_requests()
+                    else:
+                        result['donation'] = donation.to_dict()
+                    return jsonify(result), 200
+            except:
+                return jsonify(message="Server Error!"), 500
+        else:
+            return jsonify(message="Bad request!"), 400
+
+    @staticmethod
+    def get_donations_by_user(uid):
+        if uid:
+            try:
+                donations = Donation.get_donations_by_user(uid)
+                result_list = [donation.to_dict() for donation in donations]
+                result = {
+                    "message": "Success!",
+                    "donation": result_list
+                }
+                return jsonify(result), 200
+            except:
+                return jsonify(message="Server Error!"), 500
+        else:
+            return jsonify(message="Bad request!"), 400
```

Comment:
This method is not necessary, please refer to the first comment in the app routes.